tests/StartWithSetup.fs

```diff
@@ -0,0 +1,74 @@
+module Ionide.LanguageServerProtocol.Tests.StartWithSetup
+
+open Expecto
+open System.IO.Pipes
+open System.IO
+open Ionide.LanguageServerProtocol
+open Ionide.LanguageServerProtocol.Server
+
+type TestLspClient(sendServerNotification: ClientNotificationSender, sendServerRequest: ClientRequestSender) =
+    inherit LspClient ()
+
+let setupEndpoints(_: LspClient): Map<string, System.Delegate> =
+    [] |> Map.ofList
+
+let requestWithContentLength(request: string) =
```

Comment:
`let requestWithContentLength(request: string) =
    $"Content-Length: {request.Length}\r\n\r\n{request}"`

git can convert newlines in files, and since this is a string literal, it caused a problem when the underlying file bytes changed. It was coded on windows which is \r\n, but got converted to unix line endings on push which is \n only. The json rpc stream [requires \r\n](https://github.com/microsoft/vs-streamjsonrpc/blob/main/src/StreamJsonRpc/HeaderDelimitedMessageHandler.cs#LL458C17-L458C155).

Their error message is actually bugged. They don't escape the slashes in their string so it prints an actual newline where they intended to print "\r\n". I'll probably make a PR over there.

Also note it can't be a @ literal string now, or that would literally put \r\n in the string instead of escaping and putting the return and new line bytes.